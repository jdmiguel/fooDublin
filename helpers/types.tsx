import { ReactNode } from 'react';

export type InitialAppState = {
  favorites: Restaurant[];
  relatedRestaurants: Restaurant[];
  breadcrumbs: BreadcrumbsData[];
};

// Core / ui components

export enum LogoSize {
  BIG = 'big',
  SMALL = 'small',
}

export enum LoaderType {
  CIRCLE = 'circle',
  LINE = 'line',
}

export enum LoaderMode {
  DARK = 'dark',
  LIGHT = 'light',
}

export type ListItem = {
  iconSrc?: string;
  id: number;
  name: string;
  path: string;
};

export enum CardType {
  SUGGESTION = 'suggestion',
  STANDART = 'standart',
  HIGHLIGHT = 'highlight',
}

export enum CustomLinkSize {
  BIG = 'big',
  SMALL = 'small',
}

// Restaurant data

export type Timming = {
  id: string;
  day: string;
  schedule: string;
};

export enum EntityType {
  CITY = 'city',
  SUBZONE = 'subzone',
}

export type RestaurantsRequestParams = {
  entity_id: number | null;
  cuisines: number | null;
  entity_type: EntityType.CITY | EntityType.SUBZONE;
  start?: number;
  sort?: string;
  order?: string;
  q?: string;
  count?: number;
};

export type Restaurant = {
  id: number;
  imgSrc: string;
  title: string;
  content: ReactNode | string;
  route: string;
  asRoute: string;
};

export type RawRestaurant = {
  restaurant: {
    R: {
      has_menu_status: {
        delivery: number;
        takeaway: number;
      };
      res_id: number;
      is_grocery_store: boolean;
    };
    apikey: string;
    id: number;
    name: string;
    url: string;
    location: {
      address: string;
      locality: string;
      city: string;
      city_id: number;
      latitude: number;
      longitude: number;
      zipcode: string;
      country_id: number;
      locality_verbose: string;
    };
    switch_to_order_menu: number;
    cuisines: string;
    timings: string;
    average_cost_for_two: number;
    price_range: number;
    currency: string;
    highlights: string[];
    offers: any[];
    opentable_support: number;
    is_zomato_book_res: number;
    mezzo_provider: string;
    is_book_form_web_view: number;
    book_form_web_view_url: string;
    book_again_url: string;
    thumb: string;
    user_rating: {
      aggregate_rating: number;
      rating_text: string;
      rating_color: string;
      rating_obj: {
        title: {
          text: number;
        };
        bg_color: {
          type: string;
          tint: number;
        };
      };
      votes: number;
    };
    all_reviews_count: number;
    photos_url: string;
    photo_count: number;
    menu_url: string;
    featured_image: string;
    has_online_delivery: number;
    is_delivering_now: number;
    store_type: string;
    include_bogo_offers: boolean;
    deeplink: string;
    is_table_reservation_supported: number;
    has_table_booking: number;
    events_url: string;
    phone_numbers: string;
    all_reviews: {
      reviews: [
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
      ];
    };
    establishment: string[];
  };
};

export type RestaurantDetail = {
  imgSrc: string;
  thumbSrc: string;
  name: string;
  location: string;
  cuisines: string;
  timings: string;
  rating: number;
  votes: number;
  average: string;
  establishment: string;
  highlights: string[];
  phone: string;
  address: string;
};

export type Review = {
  id: number;
  userName: string;
  userImgSrc: string;
  rating: number;
  date: string;
  text: string;
};

export type RawReview = {
  review: {
    rating: number;
    review_text: string;
    id: number;
    rating_color: string;
    review_time_friendly: string;
    rating_text: string;
    timestamp: number;
    likes: number;
    user: {
      name: string;
      foodie_level: string;
      foodie_level_num: number;
      foodie_color: string;
      profile_url: string;
      profile_image: string;
      profile_deeplink: string;
    };
    comments_count: number;
  };
};

// Breadcrumbs

export enum BreadcrumbsType {
  HOME = 'home',
  SEARCH = 'search',
  DETAIL = 'detail',
  FAVORITES = 'favorite',
}

export type BreadcrumbsData = {
  text: string;
  route: string;
  asRoute: string;
  type:
    | BreadcrumbsType.HOME
    | BreadcrumbsType.SEARCH
    | BreadcrumbsType.DETAIL
    | BreadcrumbsType.FAVORITES;
};

// Actions

type AddFavoriteAction = {
  type: 'ADD_FAVORITE';
  favorite: Restaurant;
};

type DeleteFavoriteAction = {
  type: 'DELETE_FAVORITE';
  id: number;
};

type SetRelatedRestaurantsAction = {
  type: 'SET_RELATED_RESTAURANTS';
  relatedRestaurants: Restaurant[];
};

type ClearRelatedRestaurantsAction = {
  type: 'CLEAR_RELATED_RESTAURANTS';
};

type AddBreadcrumbsAction = {
  type: 'ADD_BREADCRUMBS';
  breadcrumbs: BreadcrumbsData;
};

type ReplaceBreadcrumbsAction = {
  type: 'REPLACE_BREADCRUMBS';
  index: number;
  breadcrumbs: BreadcrumbsData;
};

export type Actions =
  | AddFavoriteAction
  | DeleteFavoriteAction
  | SetRelatedRestaurantsAction
  | ClearRelatedRestaurantsAction
  | AddBreadcrumbsAction
  | ReplaceBreadcrumbsAction;
