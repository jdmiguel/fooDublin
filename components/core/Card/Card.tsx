import React, { ReactNode, forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { LazyImage } from 'react-lazy-images';

import {
  THUMB_GENERIC_SRC,
  HIGHLIGHT_GENERIC_SRC,
} from '../../../helpers/staticData';

export enum CardType {
  SUGGESTION = 'suggestion',
  STANDART = 'standart',
  HIGHLIGHT = 'highlight',
}

type CardProps = {
  imgSrc: string;
  title: string;
  content: ReactNode | string;
  type?: CardType;
  onClick?: () => void;
};

const CardTextCSS = css`
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
`;

const StandartCardImageCSS = css`
  max-width: 100px;
  height: 100%;
  min-width: 80px;
  @media only screen and (min-width: 768px) {
    width: 100px;
    min-width: 100px;
  }
`;

const SuggestionCardImageCSS = css`
  width: 30px;
  height: 30px;
  border: solid ${({ theme }) => theme.palette.LIGHT_MAX} 1px;
  display: inline-block;
  border-radius: 4px;
  -webkit-appearance: button-bevel;
  box-shadow: 1px 2px 5px #888;
`;

const HighlightCardImageCss = css`
  width: 100%;
  max-width: 350px;
  margin-bottom: 18px;
  width: 100%;
  max-width: 350px;
  margin-bottom: 18px;
  border: 1px solid ${({ theme }) => theme.palette.LIGHT_MEDIUM};
`;

const StyledCard = styled.a<{ type: CardType }>`
  width: 100%;
  max-width: ${({ type }) => (type === CardType.HIGHLIGHT ? '350px' : '600px')};
  height: 100%;
  min-height: ${({ type }) => type === CardType.STANDART && '100px'};
  overflow: hidden;
  display: flex;
  flex-direction: ${({ type }) => type === CardType.HIGHLIGHT && 'column'};
  align-items: ${({ type }) => type === CardType.SUGGESTION && 'center'};
  padding: ${({ type }) => {
    switch (type) {
      case CardType.HIGHLIGHT:
        return '15px 15px 25px';
      case CardType.SUGGESTION:
        return '0 0 0 10px';
      case CardType.STANDART:
      default:
        return '0';
    }
  }};
  cursor: pointer;
  background-color: ${({ type, theme }) =>
    type !== CardType.SUGGESTION && theme.palette.LIGHT_MAX};
  @media only screen and (min-width: 768px) {
    transition: background-color 0.2s ease-out;
    &:hover {
      background-color: ${({ theme }) => theme.palette.PRIMARY_LIGHT};
      h4 {
        color: ${({ theme }) => theme.palette.PRIMARY};
      }
    }
  }
`;

const getImageCss = (type: CardType) => {
  switch (type) {
    case CardType.SUGGESTION:
      return SuggestionCardImageCSS;
    case CardType.HIGHLIGHT:
      return HighlightCardImageCss;
    case CardType.STANDART:
    default:
      return StandartCardImageCSS;
  }
};

const StyledImage = styled(LazyImage)<{ type: CardType }>`
  ${({ type }) => getImageCss(type)}
`;

const StyledGenericThumb = styled.img<{ type: CardType }>`
  ${({ type }) => getImageCss(type)}
`;

const StyledText = styled.div<{ type: CardType }>`
  display: flex;
  flex-direction: column;
  justify-content: ${({ type }) =>
    type === CardType.HIGHLIGHT ? 'space-around' : 'space-evenly'};
  padding: ${({ type }) => type !== CardType.HIGHLIGHT && '5px 0'};
  width: ${({ type }) => type !== CardType.HIGHLIGHT && '65%'};
  margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '14px'};
  @media only screen and (min-width: 768px) {
    font-size: 1.3rem;
    line-height: 1.3rem;
    padding: ${({ type }) => type !== CardType.HIGHLIGHT && '10px 0'};
    margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '16px'};
  }
  @media only screen and (min-width: 1200px) {
    margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '20px'};
  }
`;

const StyledTitle = styled.h4<{ type: CardType }>`
  ${({ type }) => type !== CardType.HIGHLIGHT && CardTextCSS}
  font-size: ${({ type }) =>
    type === CardType.SUGGESTION ? '1.05rem' : '1.3rem'};
  font-weight: 600;
  color: ${({ theme }) => theme.palette.PRIMARY_MEDIUM};
  margin-bottom: ${({ type }) => type === CardType.HIGHLIGHT && '6px'};
  @media only screen and (min-width: 768px) {
    transition: color 0.2s ease-out;
  }
`;

const StyledContent = styled.p<{ type: CardType }>`
  ${({ type }) => type !== CardType.HIGHLIGHT && CardTextCSS}
  font-size: ${({ type }) => (type === CardType.STANDART ? '1rem' : '0.92rem')};
  line-height: 1.1rem;
  font-weight: 500;
  color: ${({ theme }) => theme.palette.DARK_MEDIUM};
`;

export const Card = forwardRef<HTMLAnchorElement, CardProps>(
  (
    { imgSrc, title, content, onClick, type = CardType.STANDART },
    forwardedRef,
  ) => (
    <StyledCard
      ref={forwardedRef}
      className={type === CardType.SUGGESTION ? '' : 'paper'}
      onClick={onClick && onClick}
      type={type}
      data-testid="card"
    >
      <StyledImage
        src={imgSrc}
        alt={title}
        type={type}
        placeholder={({ imageProps, ref }) => (
          <div ref={ref} className="LazyImage-Placeholder">
            <StyledGenericThumb
              src={
                type === CardType.HIGHLIGHT
                  ? HIGHLIGHT_GENERIC_SRC
                  : THUMB_GENERIC_SRC
              }
              alt={imageProps.alt}
              type={type}
            />
          </div>
        )}
        actual={({ imageProps }) => (
          <div className="LazyImage-Actual">
            <img {...imageProps} alt={title} />
          </div>
        )}
      />
      <StyledText type={type}>
        <StyledTitle type={type}>{title}</StyledTitle>
        <StyledContent type={type}>{content}</StyledContent>
      </StyledText>
    </StyledCard>
  ),
);

Card.displayName = 'Card';
