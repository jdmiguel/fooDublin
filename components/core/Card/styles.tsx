import styled, { css } from 'styled-components';
import { LazyImage } from 'react-lazy-images';
import { CardType } from '../types';

const CardTextCSS = css`
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
`;

const StandartCardImageCSS = css`
  width: 98px;
  height: 98px;
  object-fit: cover;
`;

const SuggestionCardImageCSS = css`
  width: 30px;
  height: 30px;
  border: solid ${({ theme }) => theme.palette.LIGHT_MAX} 1px;
  display: inline-block;
  border-radius: 4px;
  appearance: button-bevel;
  box-shadow: 1px 2px 5px ${({ theme }) => theme.palette.DARK_MIN};
`;

const HighlightCardImageCss = css`
  width: 100%;
  margin-bottom: 18px;
  border: 1px solid ${({ theme }) => theme.palette.LIGHT_MEDIUM};
`;

export const StyledCard = styled.a<{ type: CardType }>`
  width: 100%;
  height: 100%;
  min-height: ${({ type }) => type === CardType.STANDART && '100px'};
  overflow: hidden;
  display: flex;
  flex-direction: ${({ type }) => type === CardType.HIGHLIGHT && 'column'};
  align-items: ${({ type }) => type === CardType.SUGGESTION && 'center'};
  padding: ${({ type }) => {
    switch (type) {
      case CardType.HIGHLIGHT:
        return '15px 15px 25px';
      case CardType.SUGGESTION:
        return '0 0 0 10px';
      case CardType.STANDART:
      default:
        return '0';
    }
  }};
  cursor: pointer;
  background-color: ${({ type, theme }) => type !== CardType.SUGGESTION && theme.palette.LIGHT_MAX};
  @media only screen and (min-width: 768px) {
    max-width: 521px;
    transition: background-color 0.2s ease-out;
    &:hover {
      background-color: ${({ theme }) => theme.palette.PRIMARY_LIGHT};
      h4 {
        color: ${({ theme }) => theme.palette.PRIMARY};
      }
    }
  }
`;

const getImageCss = (type: CardType) => {
  switch (type) {
    case CardType.SUGGESTION:
      return SuggestionCardImageCSS;
    case CardType.HIGHLIGHT:
      return HighlightCardImageCss;
    case CardType.STANDART:
    default:
      return StandartCardImageCSS;
  }
};

export const StyledImage = styled(LazyImage)<{ type: CardType }>`
  ${({ type }) => getImageCss(type)}
`;

export const StyledGenericThumb = styled.img<{ type: CardType }>`
  ${({ type }) => getImageCss(type)}
`;

export const StyledText = styled.div<{ type: CardType }>`
  display: flex;
  flex-direction: column;
  justify-content: ${({ type }) => (type === CardType.HIGHLIGHT ? 'space-around' : 'space-evenly')};
  padding: ${({ type }) => type !== CardType.HIGHLIGHT && '5px 0'};
  width: ${({ type }) => type !== CardType.HIGHLIGHT && '65%'};
  margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '14px'};
  @media only screen and (min-width: 768px) {
    font-size: 1.3rem;
    line-height: 1.3rem;
    padding: ${({ type }) => type !== CardType.HIGHLIGHT && '10px 0'};
    margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '16px'};
  }
  @media only screen and (min-width: 1200px) {
    margin-left: ${({ type }) => type !== CardType.HIGHLIGHT && '20px'};
  }
`;

export const StyledTitle = styled.h4<{ type: CardType }>`
  ${({ type }) => type !== CardType.HIGHLIGHT && CardTextCSS}
  font-size: ${({ type }) => (type === CardType.SUGGESTION ? '1.05rem' : '1.3rem')};
  font-weight: 600;
  color: ${({ theme }) => theme.palette.PRIMARY_MEDIUM};
  margin-bottom: ${({ type }) => type === CardType.HIGHLIGHT && '6px'};
  @media only screen and (min-width: 768px) {
    transition: color 0.2s ease-out;
  }
`;

export const StyledContent = styled.p<{ type: CardType }>`
  ${({ type }) => type !== CardType.HIGHLIGHT && CardTextCSS}
  font-size: ${({ type }) => (type === CardType.STANDART ? '1rem' : '0.92rem')};
  line-height: 1.1rem;
  font-weight: 500;
  color: ${({ theme }) => theme.palette.DARK_MEDIUM};
`;
