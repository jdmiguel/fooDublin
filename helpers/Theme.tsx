import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
