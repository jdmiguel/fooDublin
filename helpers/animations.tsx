import { keyframes, css } from 'styled-components';

const fadeIn = (endValue: number) => keyframes`
  0% { opacity: 0 }
  100% { opacity: ${endValue} }
`;

const fadeInFromPosY = (initialPosY: number) => keyframes`
  0% { opacity: 0; transform: translateY(${initialPosY}px); }
  100% { opacity: 1; ; transform: translateY(0); }
`;

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const moveX = keyframes`
  0% { transform: translateX(-100%); }
  100% { transform: translateX(0); }
`;

export const fadeInAnimation = css<{ endValue?: number }>`
  animation: ${({ endValue }) => fadeIn(endValue || 1)} 0.15s linear;
`;

export const fadeInFromPosYAnimation = css<{
  initialPosY?: number;
  animationDuration?: number;
  animationDelay?: number;
}>`
  animation-name: ${({ initialPosY }) => fadeInFromPosY(initialPosY || 50)};
  animation-duration: ${({ animationDuration }) => animationDuration || 0.15}s;
  animation-delay: ${({ animationDelay }) => animationDelay || 0.25}s;
  animation-fill-mode: forwards;
  animation-timing-function: ease-out;
`;

export const circleLoaderAnimation = css`
  animation: ${rotate} 0.8s linear infinite;
`;

export const lineLoaderAnimation = css`
  animation: ${moveX} 1s ease-in-out infinite forwards;
`;
