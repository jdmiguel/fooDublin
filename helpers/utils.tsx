import { createStore, Reducer } from 'redux';
import appReducer from '@/store/redux/reducer';
import { Restaurant } from '@/components/pages/types';

// GENERAL UTILS

type ComposableStringFunction = (text: string) => string;

export const compose =
  (...fns: ComposableStringFunction[]) =>
  (value: string) =>
    fns.reduce((acc, fn) => fn(acc), value);

export const getRandomInt = (min: number, max: number) =>
  Math.floor(Math.random() * (max - min)) + min;

//const checkAlphanumeric = (name: string) => !!name.match(/^[0-9a-zA-Z]+$/);

export const getAlphanumericText = (text: string) => text.replace(/[^a-z0-9]/gi, '');

export const getLoweredText = (text: string) => text.toLowerCase();

export const getFormattedUrlText = (text: string, isPath = false) => {
  const concatenatorSymbol = isPath ? '-' : '+';
  return text.split(' ').reduce((acc: string, next: string) => {
    const concatenator = acc && getAlphanumericText(next) ? concatenatorSymbol : '';
    const formattedText = compose(getLoweredText, getAlphanumericText);

    return `${acc}${concatenator}${formattedText(next)}`;
  }, '');
};

// RELATED RESTAURANTS

export const getRandomNumbersList = (
  excludedValue: number,
  minRandomValue: number,
  maxRandomValue: number,
) => {
  const indexArray: number[] = [];

  while (indexArray.length < 3) {
    const random = getRandomInt(minRandomValue, maxRandomValue);
    if (random !== excludedValue && !indexArray.includes(random)) {
      indexArray.push(random);
    }
  }

  return indexArray;
};

export const getCurrentRelatedRestaurants = (
  restaurants: Restaurant[],
  currentRestaurantId: number,
) => {
  const currentSuggestionIndex = restaurants.findIndex(
    (restaurants) => restaurants.id === currentRestaurantId,
  );
  const getRelatedRestaurantsIndexList = getRandomNumbersList(
    currentSuggestionIndex,
    0,
    restaurants.length,
  );

  return getRelatedRestaurantsIndexList.map(
    (relatedRestaurantsIndex) => restaurants[relatedRestaurantsIndex],
  );
};

// SEARCH AND FAVORITES PAGE TITLE

export const getTitleText = (total: number) => ({
  totalText: total > 0 ? total : 'There are no',
  restaurantText: `restaurant${total === 0 || total >= 2 ? 's' : ''}`,
});

// CreateMockStore

export const createTestStore = () => createStore(appReducer as Reducer);

type GetSSRResult<TProps> = { props: TProps } | { redirect: any } | { notFound: true };

type GetSSRFn<TProps> = (args: any) => Promise<GetSSRResult<TProps>>;

// TYPES

export type inferSSRProps<TFn extends GetSSRFn<any>> = TFn extends GetSSRFn<infer TProps>
  ? NonNullable<TProps>
  : never;

type GetStaticResult<TProps> = { props: TProps } | { redirect: any } | { notFound: true };

type GetStaticFn<TProps> = (args: any) => Promise<GetStaticResult<TProps>>;

export type inferStaticProps<TFn extends GetStaticFn<any>> = TFn extends GetStaticFn<infer TProps>
  ? NonNullable<TProps>
  : never;
