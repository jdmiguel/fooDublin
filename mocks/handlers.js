import { rest } from 'msw';
import { searchInputRestaurants } from './data/search-input-restaurants';

export const handlers = [
  rest.get(`${process.env.BASE_API}search`, (req, res, ctx) => {
    return res(ctx.json(searchInputRestaurants));
  }),
];
