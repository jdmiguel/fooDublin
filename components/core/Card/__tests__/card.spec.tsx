import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Card }, { CardType } from '../Card';

import { CARD_PROPS_MOCKS } from '../__mocks__/card.mocks';

import { renderWithTheme } from '../../../../helpers/Theme';
import { CDN_URL_STATIC_DIRECTORY } from '../../../../helpers/utils';

it('should render standart card', () => {
  const { container } = render(
    renderWithTheme(
      <Card
        {...CARD_PROPS_MOCKS}
        imgSrc={`${CDN_URL_STATIC_DIRECTORY}/images/thumb-1.webp`}
      />,
    ),
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('should render suggestion card', () => {
  const { container } = render(
    renderWithTheme(
      <Card
        {...CARD_PROPS_MOCKS}
        imgSrc={`${CDN_URL_STATIC_DIRECTORY}/images/thumb-1.webp`}
        type={CardType.SUGGESTION}
      />,
    ),
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('should render highlight card', () => {
  const { container } = render(
    renderWithTheme(
      <Card
        {...CARD_PROPS_MOCKS}
        imgSrc={`${CDN_URL_STATIC_DIRECTORY}/images/cleaver-east.jpg`}
        type={CardType.HIGHLIGHT}
      />,
    ),
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('should call function on click', () => {
  const handleClick = jest.fn();
  const { getByTestId } = render(
    renderWithTheme(
      <Card
        {...CARD_PROPS_MOCKS}
        imgSrc={`${CDN_URL_STATIC_DIRECTORY}/images/thumb-1.webp`}
        onClick={handleClick}
      />,
    ),
  );

  fireEvent.click(getByTestId('card'));

  expect(handleClick).toHaveBeenCalled();
});
