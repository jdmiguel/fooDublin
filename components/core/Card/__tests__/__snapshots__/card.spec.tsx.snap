// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render highlight card 1`] = `
.c0 {
  width: 100%;
  max-width: 350px;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px 15px 25px;
  cursor: pointer;
  background-color: #FFFFFF;
}

.c1 {
  width: 100%;
  max-width: 350px;
  margin-bottom: 18px;
  width: 100%;
  max-width: 350px;
  margin-bottom: 18px;
  border: 1px solid #F3F3F3;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c3 {
  font-size: 1.3rem;
  font-weight: 600;
  color: #FBA52B;
  margin-bottom: 6px;
}

.c4 {
  font-size: 0.92rem;
  line-height: 1.1rem;
  font-weight: 500;
  color: #585858;
}

@media only screen and (min-width:768px) {
  .c0 {
    -webkit-transition: background-color 0.2s ease-out;
    transition: background-color 0.2s ease-out;
  }

  .c0:hover {
    background-color: #FDF3E5;
  }

  .c0:hover h4 {
    color: #EC7E00;
  }
}

@media only screen and (min-width:768px) {
  .c2 {
    font-size: 1.3rem;
    line-height: 1.3rem;
  }
}

@media only screen and (min-width:1200px) {

}

@media only screen and (min-width:768px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }
}

<a
  class="c0 paper"
  data-testid="card"
  type="highlight"
>
  <div
    class="LazyImage-Placeholder"
  >
    <img
      alt="Default title"
      class="c1"
      src="http://localhost:3000/static/images/highlight-generic.png"
      type="highlight"
    />
  </div>
  <div
    class="c2"
    type="highlight"
  >
    <h4
      class="c3"
      type="highlight"
    >
      Default title
    </h4>
    <p
      class="c4"
      type="highlight"
    >
      Default content
    </p>
  </div>
</a>
`;

exports[`should render standart card 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  height: 100%;
  min-height: 100px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  cursor: pointer;
  background-color: #FFFFFF;
}

.c1 {
  max-width: 100px;
  height: 100%;
  min-width: 80px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  padding: 5px 0;
  width: 65%;
  margin-left: 14px;
}

.c3 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
  font-size: 1.3rem;
  font-weight: 600;
  color: #FBA52B;
}

.c4 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
  font-size: 1rem;
  line-height: 1.1rem;
  font-weight: 500;
  color: #585858;
}

@media only screen and (min-width:768px) {
  .c0 {
    -webkit-transition: background-color 0.2s ease-out;
    transition: background-color 0.2s ease-out;
  }

  .c0:hover {
    background-color: #FDF3E5;
  }

  .c0:hover h4 {
    color: #EC7E00;
  }
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:768px) {
  .c1 {
    width: 100px;
    min-width: 100px;
  }
}

@media only screen and (min-width:768px) {
  .c2 {
    font-size: 1.3rem;
    line-height: 1.3rem;
    padding: 10px 0;
    margin-left: 16px;
  }
}

@media only screen and (min-width:1200px) {
  .c2 {
    margin-left: 20px;
  }
}

@media only screen and (min-width:768px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }
}

<a
  class="c0 paper"
  data-testid="card"
  type="standart"
>
  <div
    class="LazyImage-Placeholder"
  >
    <img
      alt="Default title"
      class="c1"
      src="http://localhost:3000/static/images/generic-thumb.png"
      type="standart"
    />
  </div>
  <div
    class="c2"
    type="standart"
  >
    <h4
      class="c3"
      type="standart"
    >
      Default title
    </h4>
    <p
      class="c4"
      type="standart"
    >
      Default content
    </p>
  </div>
</a>
`;

exports[`should render suggestion card 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0 0 10px;
  cursor: pointer;
}

.c1 {
  width: 30px;
  height: 30px;
  border: solid #FFFFFF 1px;
  display: inline-block;
  border-radius: 4px;
  -webkit-appearance: button-bevel;
  box-shadow: 1px 2px 5px #888;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  padding: 5px 0;
  width: 65%;
  margin-left: 14px;
}

.c3 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
  font-size: 1.05rem;
  font-weight: 600;
  color: #FBA52B;
}

.c4 {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 92%;
  font-size: 0.92rem;
  line-height: 1.1rem;
  font-weight: 500;
  color: #585858;
}

@media only screen and (min-width:768px) {
  .c0 {
    -webkit-transition: background-color 0.2s ease-out;
    transition: background-color 0.2s ease-out;
  }

  .c0:hover {
    background-color: #FDF3E5;
  }

  .c0:hover h4 {
    color: #EC7E00;
  }
}

@media only screen and (min-width:768px) {
  .c2 {
    font-size: 1.3rem;
    line-height: 1.3rem;
    padding: 10px 0;
    margin-left: 16px;
  }
}

@media only screen and (min-width:1200px) {
  .c2 {
    margin-left: 20px;
  }
}

@media only screen and (min-width:768px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }
}

<a
  class="c0"
  data-testid="card"
  type="suggestion"
>
  <div
    class="LazyImage-Placeholder"
  >
    <img
      alt="Default title"
      class="c1"
      src="http://localhost:3000/static/images/generic-thumb.png"
      type="suggestion"
    />
  </div>
  <div
    class="c2"
    type="suggestion"
  >
    <h4
      class="c3"
      type="suggestion"
    >
      Default title
    </h4>
    <p
      class="c4"
      type="suggestion"
    >
      Default content
    </p>
  </div>
</a>
`;
